logging/nxscope/nxscope_chan.c

```diff
@@ -0,0 +1,1566 @@
+   *   how much memory is needed for the channel with the longest channel
+   *   name. Or just allocate memor for CHAN_NAMELEN_MAX ?
+   */
+
+  txbuf_len = (sizeof(struct nxscope_chinfo_s) +
+               strnlen(name, CHAN_NAMELEN_MAX) + 1);
+  ret = nxscope_txbuf_realloc(s, txbuf_len);
+  if (ret < 0)
+    {
+      _err("ERROR: txbuf realloc failed %d\n", ret);
+      goto errout;
+    }
+
+  /* Reset channel data */
+
+  memset(&s->chinfo[ch], 0, sizeof(struct nxscope_chinfo_s));
+
+  /* Copy channel info */
+
+  s->chinfo[ch].type.u8 = type;
+  s->chinfo[ch].vdim    = vdim;
+  s->chinfo[ch].mlen    = mlen;
+  s->chinfo[ch].name    = name;
+
+  nxscope_unlock(s);
+
+errout:
+  return ret;
+}
+
+/****************************************************************************
+ * Name: nxscope_chan_en
+ *
+ * Description:
+ *   Enable/disable a given channel
+ *
+ * Input Parameters:
+ *   s  - a pointer to a nxscope instance
+ *   ch - a channel id
+ *   en - enable/disable
+ *
+ ****************************************************************************/
+
+int nxscope_chan_en(FAR struct nxscope_s *s, uint8_t ch, bool en)
+{
+  int ret = OK;
+
+  DEBUGASSERT(s);
+
+  nxscope_lock(s);
+
+  if (ch > s->cmninfo.chmax)
+    {
+      _err("ERROR: invalid channel %d\n", ch);
+      ret = -EINVAL;
+      goto errout;
+    }
+
+  _info("chan_en=%d %d\n", ch, en);
+
+  /* Set enable flag */
+
+  s->chinfo[ch].enable = en;
+
+errout:
+  nxscope_unlock(s);
+
+  return ret;
+}
+
+#ifdef CONFIG_LOGGING_NXSCOPE_DIVIDER
+/****************************************************************************
+ * Name: nxscope_chan_div
+ *
+ * Description:
+ *   Configure divider for a given channel
+ *
+ * Input Parameters:
+ *   s   - a pointer to a nxscope instance
+ *   ch  - a channel id
+ *   div - divider value - starts from 0
+ *
+ ****************************************************************************/
+
+int nxscope_chan_div(FAR struct nxscope_s *s, uint8_t ch, uint8_t div)
+{
+  int ret = OK;
+
+  DEBUGASSERT(s);
+
+  nxscope_lock(s);
+
+  if (ch > s->cmninfo.chmax)
+    {
+      _err("ERROR: invalid channel %d\n", ch);
+      ret = -EINVAL;
+      goto errout;
+    }
+
+  _info("chan_div=%d %d\n", ch, div);
+
+  /* Set divider */
+
+  s->chinfo[ch].div = div;
+
+errout:
+  nxscope_unlock(s);
+
+  return ret;
+}
+#endif
+
+/****************************************************************************
+ * Name: nxscope_chan_all_en
+ *
+ * Description:
+ *   Enable/disable all channels
+ *
+ * Input Parameters:
+ *   s  - a pointer to a nxscope instance
+ *   en - enable/disable
+ *
+ ****************************************************************************/
+
+int nxscope_chan_all_en(FAR struct nxscope_s *s, bool en)
+{
+  int ret = OK;
+  int i   = 0;
+
+  DEBUGASSERT(s);
+
+  for (i = 0; i < s->cmninfo.chmax; i++)
+    {
+      ret = nxscope_chan_en(s, i, en);
+      if (ret < 0)
+        {
+          goto errout;
```

Comment:
Are you sure you want to exit early when one channel fails to enable/disable? Wouldn't it be better to disable/enable all channels, but return error if at least one channel failed? `nxscope_chan_en` returns only one error so you could do `ret |= nxscope_chan_en(s, i, en);` and skip the `if (ret < 0)`

Reply:
I don't have any strong opinion here. But for sure what you propose will save us a few lines so I changed it.